### PR TITLE
Allow custom error messages to be passed in filter spec

### PR DESCRIPTION
**Pull request by chadicus**
Allows user to specify custom error message when a value fails to filter.

### CODE REVIEW

**Comment by chrisryan** [comment]
:+1:

**Comment by chrisryan** [comment]
I noticed that the php7 version failed on Travis but the reason is not because of the changes in this PR. The most recent tag 2.0 is supposed to include php7 support but something is obviously wrong.

File: tests/FiltererTest.php [hunk 1]
```diff
@@ -429,4 +429,65 @@ public static function passingFilter($value)
...
+     * @test
+     * @covers ::filter
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage error for field 'fieldOne' was not a non-empty string
+     *
+     * @return void
+     */
+    public function filterWithNonStringError()
+    {
+        F::filter(
```
Comment (tmli3b3rm4n) [inline tests/FiltererTest.php]: `use DominionEnterprises\Filterer as F;`  I know this wasn't introduced in this PR.  I would get rid of `as F` and do a find and replace all instances of `F` to Filterer.

Comment (chrisryan) [inline tests/FiltererTest.php]: I agree that would be a good change in a separate PR.
